### ISSUE TITLE
Detection of active runtime version manager fails

### ISSUE THREAD

**Issue by Terbium-135** [lexical-lsp/lexical#440]
Checked out lexical main up to:

```
commit cb138a8ad492e52448b615b33d9a0687ad8f926e (HEAD -> main, origin/main, origin/HEAD)
Author: Zach Allaun <zach.allaun@gmail.com>
Date:   Fri Oct 20 10:44:35 2023 -0700
```

I am unable to start lexical (editor is `vscode` here):
```
Starting lexical release in "/home/elixir-dev/projects/lexical/_build/dev/package/lexical/bin"
[Trace - 08:59:21] Sending request 'initialize - (0)'.
No activated version manager detected. Searching for version manager...
Found asdf. Activating...
Detected Elixir through asdf: /home/elixir-dev/.asdf/installs/elixir/1.15.7-otp-26/bin/elixir
[Error - 08:59:34] Client Lexical: connection to server is erroring. Shutting down server.
[Error - 08:59:34] Stopping server failed
Error: Client is not running and can't be stopped. It's current state is: starting
```

I have both runtime managers installed `asdf` as well as `rtx` but **only rtx is activated** in `.zshrc`:
```
# . $HOME/.asdf/asdf.sh
# . $HOME/.asdf/completions/asdf.bash
eval "$(~/.local/share/rtx/bin/rtx activate zsh)"
```

After manually *hiding* the `.asdf` folder lexical starts normal

Let me know if further information is needed

**Comment by zachallaun** [comment]
Out of curiosity, how are you starting VS Code? Do you start it using `code` from the command line or do you start the app from a start menu/desktop?

**Comment by zachallaun** [comment]
Also, to test a theory, would you mind adding the rtx activation to `.bashrc` and copy/paste the same startup logs from lexical?

```
# ~/.bashrc
eval "$(~/.local/share/rtx/bin/rtx activate bash)"
```

**Comment by Terbium-135** [comment]
VS Code is running on Windows and remotely connecting to the linux computer with the source

`.bashrc` already has rtx added:
```
. "$HOME/.asdf/asdf.sh"
. "$HOME/.asdf/completions/asdf.bash"
eval "$(/home/elixir-dev/.local/share/rtx/bin/rtx activate bash)"
```

but `asdf` never got commented out since I am not using `bash` anymore but `zsh`

I don't know the detection logic you implemented but perhaps it should respect the active shell?
Just guessing here...

**Comment by zachallaun** [comment]
> VS Code is running on Windows and remotely connecting to the linux computer with the source

Yep, I use the same setup. However, could you clarify whether VS Code is started using `code` from the terminal or through the Windows start menu? If from the terminal, I believe VS Code inherits the environment present in the shell it was launched from. There's definitely an issue in the detection logic when both asdf and rtx are found, but I'm trying to narrow a few things down now because I'll only have a little bit of time to work on it later today.

Could you also please clarify the behavior if you do not initialize asdf in your bashrc and only initialize rtx?

**Comment by Terbium-135** [comment]
I'm starting VS Code from an icon on my windows taskbar.

I'm using zsh, but despite of that: in both files (`.zshrc` and `.bashrc`) asdf is commented out now.
As soon as the .asdf folder is in my home directory it gets detected...
```
Starting lexical release in "/home/elixir-dev/projects/lexical/_build/dev/package/lexical/bin"
[Trace - 19:01:29] Sending request 'initialize - (0)'.
No activated version manager detected. Searching for version manager...
Found asdf. Activating...
Detected Elixir through asdf: /home/elixir-dev/.asdf/installs/elixir/1.15.7-otp-26/bin/elixir
[Error - 19:01:42] Client Lexical: connection to server is erroring. Shutting down server.
[Error - 19:01:42] Stopping server failed
Error: Client is not running and can't be stopped. It's current state is: starting
```

Please don't spend to much time on this edge case
Thank you for qll the work!

**Comment by zachallaun** [comment]
Should be fixed on main and in the 0.4.0 release!
